### ISSUE TITLE
Basis ordering discrepancy between Rigetti and staq

### ISSUE THREAD

**Issue by DevelopDaily** [softwareQinc/staq#36]
Here is a simple test case:

```
OPENQASM 2.0;
include "qelib1.inc";

qreg q[2];
id q[0];
x q[1];
```

The final state by a `qpp `run:

```
>> Final state:
0   1   0   0 

```
The snippet of the `Rigetti Quil` code out of the `staq`:

```
I 0
X 1
```

The final state by a `Rigetti `run:

```
[
0.+0.j 
0.+0.j 
1.+0.j 
0.+0.j
]
```

`Rigetti `makes a very big deal of basis ordering, which is documented [here](https://pyquil-docs.rigetti.com/en/stable/wavefunction_simulator.html).

I quote the key point:

> The WavefunctionSimulator enumerates bitstrings such that qubit 0 is the least significant bit (LSB)...
> 
> This convention is counter to that often found in the quantum computing literature where bitstrings are often ordered such that the lowest-index qubit is on the left.

**Comment by vsoftco** [comment]
I think `staq` does the right thing in simply translating gate sets. The qubit ordering comes into play only when executing. In other words, the assembly code should be invariant, but execution will depend of course on how those bits are interpreted. I am very much tempted to leave staq as is, and only document the endianness (bit ordering) on the simulator part. That's because Rigetti still numbers qubits as we do, but only "interpret" them differently when writing down the state vector. 

So, in other words, I still want 

    x q[0] // qasm

to be translated to 

    X 0 // Rigetti

and deal with endianness at the end of processing (i.e., when executing on a simulator). Otherwise, generating code with bits reversed will create a lot of misunderstandings and further issues down the line.

**Comment by DevelopDaily** [comment]
Agreed. Well-thought-out. As long as it is a conscious design decision and well documented, that should be good enough.

Or, if possible, a utility standalone function to prepare/interpret endianness would be nice, leaving the critical `QEngine ` intact. If impossible, the current code can be kept as it is.

**Comment by vsoftco** [comment]
@DevelopDaily Good point, will think about it, let's keep this open. I'll refer this to https://github.com/softwareQinc/qpp/issues/104, as they are closely related

**Comment by DevelopDaily** [comment]
Great. Thanks.
